### PR TITLE
generate friendship controller, user can stop following & followed friends shows as followed

### PR DESCRIPTION
**Pull request by LeeRenJie**
depends on #139

### CODE REVIEW

File: finance-tracker/app/controllers/friendships_controller.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class FriendshipsController < ApplicationController
+  def create
```
Comment (tlkiong) [inline finance-tracker/app/controllers/friendships_controller.rb:2]: Whats this for?
If no use, remove it and don't forget to remove it from route as well

File: finance-tracker/app/controllers/friendships_controller.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  end
+
+  def destroy
+    friendship = current_user.friendships.find_by(friend_id: params[:id])
+    friendship.destroy
+    flash[:notice] = "Stopped following"
```
Comment (tlkiong) [inline finance-tracker/app/controllers/friendships_controller.rb:8]: How sure are you the `.destroy` happen without any error?
You doing it like this is:

```
1. destroy friendship
2. error in destroying friendship
3. Set flash notice to say "Stopped following". Then refresh page the data still tehre case of (2)
```

File: finance-tracker/app/models/user.rb [hunk 3]
```diff
@@ -50,4 +50,8 @@ def self.matches(field_name, param)
   def except_current_user(users)
     users.reject { |user| user.id == self.id }
   end
+
+  def not_friends_with?(friend_id)
+    !self.friends.where(id: friend_id).exists?
```
Comment (tlkiong) [inline finance-tracker/app/models/user.rb:55]: can you use the `exists` code I gave you previously?

**Review by tlkiong — CHANGES_REQUESTED** [review]

File: finance-tracker/app/controllers/friendships_controller.rb (hunk 1 above)
Comment (LeeRenJie) [reply finance-tracker/app/controllers/friendships_controller.rb:2]: it will be used in the next PR

Comment (tlkiong) [reply finance-tracker/app/controllers/friendships_controller.rb:2]: Next time, have it in the next PR instead of here

**Review by tlkiong — APPROVED** [review]

**Pull request merged**